### PR TITLE
drivers: sensor_sim: Fix sine calculation

### PR DESCRIPTION
**Pull request by MarekPieta**
Change fixes sine calculation. The argument should be of double type and represent angle in radians.

### CODE REVIEW

File: drivers/sensor/sensor_sim/sensor_sim.c [hunk 1]
```diff
@@ -211,9 +214,13 @@ static double generate_pseudo_random(void)
  */
 static double generate_sine(double offset, double amplitude)
 {
-	uint32_t time = k_uptime_get_32();
+	static const double angle_max = 2 * M_PI;
+	static const uint32_t period_ms = 10000;
 
-	return offset + amplitude * sin(time % 65535);
+	double time = (double)(k_uptime_get_32() % period_ms);
+	double angle = (time / (double)period_ms) * angle_max;
```
[the remark this answers is not in the document]
Comment (pdunaj) [reply drivers/sensor/sensor_sim/sensor_sim.c:221]: I would prefer to use `2*PI` directly without the `angle_max`. I guess everybody knows what 2pi is (especially in contect of generating a sine wave) and angle_max can create a confusion that there is something that can actually be controlled (like period)

**Review by pdunaj — CHANGES_REQUESTED** [review]

File: drivers/sensor/sensor_sim/sensor_sim.c [hunk 2]
```diff
@@ -211,9 +214,13 @@ static double generate_pseudo_random(void)
  */
 static double generate_sine(double offset, double amplitude)
 {
-	uint32_t time = k_uptime_get_32();
+	static const double angle_max = 2 * M_PI;
+	static const uint32_t period_ms = 10000;
 
-	return offset + amplitude * sin(time % 65535);
+	double time = (double)(k_uptime_get_32() % period_ms);
```
[the remark this answers is not in the document]
Comment (MarekPieta) [reply drivers/sensor/sensor_sim/sensor_sim.c:220]: Removed

File: drivers/sensor/sensor_sim/sensor_sim.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarekPieta) [reply drivers/sensor/sensor_sim/sensor_sim.c:221]: done

File: drivers/sensor/sensor_sim/sensor_sim.c [hunk 3]
```diff
@@ -211,9 +214,12 @@ static double generate_pseudo_random(void)
  */
 static double generate_sine(double offset, double amplitude)
 {
-	uint32_t time = k_uptime_get_32();
+	static const uint32_t period_ms = 10000;
 
-	return offset + amplitude * sin(time % 65535);
+	double time = (k_uptime_get_32() % period_ms);
```
Comment (pdunaj) [inline drivers/sensor/sensor_sim/sensor_sim.c:219]: once the cast is gone there is also no need to brackets

**Review by pdunaj — APPROVED** [review]

File: drivers/sensor/sensor_sim/sensor_sim.c (hunk 3 above)
Comment (MarekPieta) [reply drivers/sensor/sensor_sim/sensor_sim.c:219]: done

**Comment by pdunaj** [comment]
Hi @anangl , this PR is ready too

**Review by anangl — APPROVED** [review]

**Review by jtguggedal — APPROVED** [review]

**Pull request merged**